api/src/main/java/ca/bc/gov/educ/api/trax/service/institute/DistrictService.java

```diff
@@ -58,4 +64,33 @@ public void loadDistrictsIntoRedisCache(List<District> districts) {
                 .saveAll(districtTransformer.transformToEntity(districts));
     }
 
+    public List<District> getSchoolsFromRedisCache() {
+        log.debug("**** Getting districts from Redis Cache.");
+        return  districtTransformer.transformToDTO(districtRedisRepository.findAll());
+    }
+
+    public void initializeDistrictCache(boolean force) {
+        String cacheStatus = redisTemplate.opsForValue().get(CacheKey.DISTRICT_CACHE.name());
+        cacheStatus = cacheStatus == null ? "" : cacheStatus;
+        if (CacheStatus.LOADING.name().compareToIgnoreCase(cacheStatus) == 0
+                || CacheStatus.READY.name().compareToIgnoreCase(cacheStatus) == 0) {
+            log.info(String.format("DISTRICT_CACHE status: %s", cacheStatus));
+            if (force) {
+                log.info("Force Flag is true. Reloading DISTRICT_CACHE...");
+                redisTemplate.opsForValue().set(CacheKey.DISTRICT_CACHE.name(), CacheStatus.LOADING.name());
```

Comment:
Duplicate code blocks in if(force) and else statements

Reply:
Fixed here: https://github.com/bcgov/EDUC-GRAD-TRAX-API/blob/c79bdcc841ed4529314fd9ef0d3bd821fb053b15/api/src/main/java/ca/bc/gov/educ/api/trax/service/institute/ServiceHelper.java#L24
Added new generic class ServiceHelper used by other institute service classes.